oneanddone/tasks/static/js/tasklist.js

```diff
@@ -0,0 +1,415 @@
+    let typeNodes = this.props.types.map(function(type) {
+      if (this.state.expanded || $.inArray(type.id.toString(), this.props.filters.type) > -1) {
+        return (
+          <Type id={type.id}
+                name={type.name}
+                key={type.id}
+            {...this.props} />
+        );
+      }
+    }, this);
+    return (
+      <div>
+        <span className="filter-heading">Types</span>
+        <span className="filter-toggle" onClick={this.handleToggle}> {(this.state.expanded) ? '-':'+'} </span>
+        <ul>
+          {typeNodes}
+        </ul>
+      </div>
+    );
+  }
+});
+
+let Type = React.createClass({
+  handleChange: function(e) {
+    let type = this.props.filters.type;
+    if (e.target.checked) {
+      type.push(e.target.value);
+    } else {
+      type.splice($.inArray(e.target.value, type), 1);
+    }
+    this.props.onFilterChange();
+  },
+  render: function() {
+    let typeId = 'type' + this.props.id;
+    return (
+      <li>
+        <label htmlFor={typeId}>
+          <input id={typeId} value={this.props.id} type="checkbox" onClick={this.handleChange} />
+          {this.props.name}
+        </label>
+      </li>
+    );
+  }
+});
+
+let TaskList = React.createClass({
+  render: function() {
+    let taskNodes = this.props.data.map(function(task) {
+      return (
+        <Task name={task.name}
+              short_description={task.short_description}
+              url={task.url}
+              edit_url={task.edit_url}
+              key={task.id}>
+          {task.short_description}
+        </Task>
+      );
+    });
+    let loadMore;
+    if (this.props.moreTasks) {
+      loadMore = <LoadMoreButton onLoadMore={this.props.onLoadMore} />
+    }
+    return (
+      <div className="task-list-container">
+        <ol className="task-list">
+          {taskNodes}
+        </ol>
+        {loadMore}
+      </div>
+    );
+  }
+});
+
+let LoadMoreButton = React.createClass({
+  render: function() {
+    return (
+      <a className="button" onClick={this.props.onLoadMore}>Load more</a>
+    );
+  }
+})
+
+let Task = React.createClass({
+  render: function() {
+    console.log('rendering task');
+    let editLink;
+    if (this.props.edit_url) {
+      editLink = <EditLink edit_url={this.props.edit_url} />
+    }
+    return (
+      <li>
+        <p>
+          <a className="task-name" href={this.props.url}>{this.props.name}</a>
+          {editLink}
+        </p>
+        <p className="task-desc">{this.props.children.toString()}</p>
+      </li>
+    );
+  }
+});
+
+let EditLink = React.createClass({
+  render: function() {
+    return (
+      <span> | <a className="edit-task" href={this.props.edit_url}>Edit</a></span>
+    );
+  }
+})
+
+
+ReactDOM.render(
+    <TaskPage url="/api/v1/task"
+              team_url="/api/v1/taskteam"
+              project_url="/api/v1/taskproject"
+              type_url="/api/v1/tasktype" />,
+    document.getElementById('content')
+);
```

Comment:
Not sure if this would get any easier to read if you moved from one major React component to a few smaller ones:

```js
ReactDOM.render(
  <div className="taskPage">
    <div className="task-filters">
      <form method="get" className="filters">
        <fieldset>
          <SearchFilter />
          <ETFilter executionTimes={[15, 30, 45, 60]} />
          <TeamFilter type_url="/api/v1/taskteam" />
          <ProjectFilter type_url="/api/v1/taskproject" />
          <TypeFilter type_url="/api/v1/tasktype" />
        </fieldset>
      </form>
    </div>

    <TaskList data={this.state.data} onLoadMore={this.loadMoreTasks} moreTasks={this.state.moreTasks} />
  </div>,

  document.getElementById('content')
);
```

Not sure if that'd work, or require massive refactoring, but it'd move props closer to their intended targets so you wouldn't need to use a massive `componentDidMount()` to load all your AJAX data.
Now each filter could use it's own `componentDidMount` cycle to load its data and hopefully remove a few of the `{...this.props}` proxies. It may also require a bit less global `this.setState()` calls and move state management into the individual components instead of the global component.